utils/hotspots.ts

```diff
@@ -100,3 +103,84 @@ export const deleteHotspot = async (
 
   return { success: true };
 };
+
+export const searchLocations = async (
+  address: string,
+  proximityPolicy: 'ip' | 'coordinate' | 'none'
+) => {
+  try {
+    const requestHeaders: HeadersInit = new Headers();
+    requestHeaders.set('Content-Type', 'application/json');
+    const server = mapboxGeocodingServerAddress;
+    const search_query = encodeURI(address);
+    const access_token = mapboxApiKey;
+
+    const response = await fetch(
+      `${server}/forward?q=${search_query}&proximity=${proximityPolicy}&access_token=${access_token}`,
+      {
+        method: 'GET',
+        headers: requestHeaders,
+      }
+    );
+
+    if (!response.ok) {
+      const errorText = await response.text();
+      const isNotSignedIn = response.status === 401;
+
+      return { error: errorText || response, isNotSignedIn };
+    }
+
+    const data = await response.json();
+    const results: LocationItemProps[] = data['features']?.map((x: any) => ({
+      place_name: x.properties.name + ', ' + x.properties.place_formatted,
+      coordinates: {
+        latitude: x.properties.coordinates.latitude,
+        longitude: x.properties.coordinates.longitude,
+      },
+    }));
```

Comment:
let's not mix snake_case with camelCase; this will make the code less predictible.
Also, can you infer the type of `x` so it won't be annotated as `any` ?

Reply:
Unfortunately, I can't really infer the type easily, but I could create a custom type that follows the same structure as the response retrieved from the MapBox API.

Reply:
Creating a custom type is alright, but to avoid using snake case, I have to create a function that maps the response to an entity containing only camel-case keys in the object. Is that necessary?

Reply:
Alright, I did it anyways (`any` types are too ambiguous, so a small amount of redundant code to convert the response to a type with camel-cased keys is not a concern).

Reply:
@sutgeorge i meant avoiding the use of snake_case in **our** code.
We have no control over how third party libraries format and case their code, so the best we can do is formatting the results of their API calls in our code.

I'm talking about the `place_name` field.